### DISCUSSION TITLE
FR: External SPort Sensors injected into telemetry

### DISCUSSION THREAD

**Discussion by wx4cb** [iNavFlight/inav#7369] [Ideas]
Not sure if this is implemented, is it possible to use a regular FRSKy Sport sensor, such as vario or even an arduino mimicing a sensor (eg: https://github.com/RealTadango/FrSky ) as an input on a serial port and send it out of the regular telemetry stream as a telemetry item?

i don't know if any processing would need to be done to it or whether it can be sent directly out as part of the telemetry stream.

Reason: I'm building a bunch of arduino based environmental measuring sensors and would rather be able to send it down as part of the telemetry as opposed to just storing it onboard and then waiting to see where i was when values i want....

for example, smoke particle measurements

**Comment by DzikuVx** [comment]
@shellixyz isn't this the `smartport master` functionality?

**Comment by wx4cb** [comment]
Im not sure that was why i was aski. Hopefully shellixyz can answer.

**Comment by shellixyz** [comment]
It is possible to connect any SPort sensor to an UART configured as "SPort master". It is emulating a receiver and allows to decode sensors' data to be used by INAV. Currently FLVSS data, vario data (altitude + vertical speed) and VS600 VTX data are decoded. These values are not currently used though. The data is also forwarded to S.Port telemetry so it is like you have them connected on the receiver. It also works the other way. S.Port commands are forwarded from receiver to S.Port master to be able to configure sensors/peripherals from the radio

**Comment by wx4cb** [comment]
@shellixyz ok so if i have a sensor that's not one of those you listed, will it be passed through to the transmitter as a normal sensor?

Will the same work if im using crossfire? will it show up as an extra sensor ?

**Comment by shellixyz** [comment]
The data sent by any sensor connected the to the SPort master port is forwarded. It is only forwarded to SPort telemetry on the receiver side though. It won't work with Crossfire

**Comment by wx4cb** [comment]
> The data sent by any sensor connected the to the SPort master port is forwarded. It is only forwarded to SPort telemetry on the receiver side though. It won't work with Crossfire

bummer, so it wouldn't show up as a regular inav sensor ?

**Comment by shellixyz** [comment]
No, only if you're using a receiver with S.Port/F.Port and it wouldn't show up as an INAV sensor, it would show up just like if you connected it directly to the receiver. What this driver brings is the possibility to talk to the sensors with INAV while still have them showing on the radio as if they were directly connected to the receiver.
